core/trino-main/src/test/java/io/trino/sql/planner/TestUnwrapCastInComparison.java

```diff
@@ -343,19 +346,19 @@ public void testDistinctFrom()
     @Test
     public void testNull()
     {
-        testUnwrap("smallint", "a = CAST(NULL AS DOUBLE)", "CAST(NULL AS BOOLEAN)");
```

Comment:
Why did these test cases change?

Reply:
After the change, predicates are not turned into `CAST(NULL AS BOOLEAN)`. Instead - there is no filter

Reply:
"there is no filter" means all rows retained
"null filter" means all rows dropped

am i missing something?

Reply:
no `FilterNode`, just a `ValuesNode` with 0 rows